server/service/src/invoice/outbound_shipment/generate.rs

```diff
@@ -0,0 +1,44 @@
+use repository::db_diesel::InvoiceLineRowType;
+use repository::{InvoiceLineRow, InvoiceRow, ItemRowRepository, RepositoryError};
+use util::uuid::uuid;
+
+use crate::service_provider::ServiceContext;
+
+pub fn generate_invoice_lines(
+    ctx: &ServiceContext,
+    invoice_row: &InvoiceRow,
+    item_ids: Vec<String>,
+) -> Result<Vec<InvoiceLineRow>, RepositoryError> {
+    let mut result: Vec<InvoiceLineRow> = Vec::new();
+
+    item_ids.into_iter().for_each(|item_id| {
+        match ItemRowRepository::new(&ctx.connection).find_one_by_id(&item_id) {
+            Ok(Some(item)) => {
+                result.push(InvoiceLineRow {
+                    id: uuid(),
+                    invoice_id: invoice_row.id.clone(),
+                    item_id: item.id.clone(),
+                    item_name: item.name.clone(),
+                    item_code: item.code.clone(),
+                    stock_line_id: None,
+                    location_id: None,
+                    batch: None,
+                    expiry_date: None,
+                    pack_size: 1,
+                    cost_price_per_pack: 0.0,
+                    sell_price_per_pack: 0.0,
+                    total_before_tax: 0.0,
+                    total_after_tax: 0.0,
+                    tax: None,
+                    r#type: InvoiceLineRowType::StockOut,
```

Comment:
Shouldn't this be UnallocatedStock ? stock_line_id must not be `None` for StockOut type (otherwise we have an error when invoice status is changed, I think when it's changed to Shipped or Allocated). Hmm I don't think this is documented anywhere (potentially we should Enforce it in repository)